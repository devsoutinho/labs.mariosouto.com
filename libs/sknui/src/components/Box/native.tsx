import React from "react";
import { View } from "react-native";
import { StyleSheet } from "../../theme/StyleSheet/native";

interface BoxProps {
  styleSheet?: StyleSheet;
  children: React.ReactNode;
}
export function Box({ styleSheet, children }: BoxProps) {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  return <View style={styleSheet as any}>{children}</View>;
}
