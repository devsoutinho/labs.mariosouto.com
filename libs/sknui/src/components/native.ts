export { Box } from "./Box/native";
export { Text } from "./Text/native";
export { Touchable } from "./Touchable/native";
export { Provider } from "./Provider/native";
