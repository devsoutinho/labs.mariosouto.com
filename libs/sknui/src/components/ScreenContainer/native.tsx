import { ScrollView } from "react-native";
import { Box } from "../Box/native";

interface ScreenContainer {
  children: React.ReactNode;
  styleSheet?: StyleSheet;
}
export function ScreenContainer({ children, styleSheet }: ScreenContainer) {
  return (
    <Box styleSheet={{ flex: 1, flexDirection: "column", ...styleSheet }}>
      <ScrollView contentContainerStyle={{ flexGrow: 1 }}>
        {children}
      </ScrollView>
    </Box>
  );
}
