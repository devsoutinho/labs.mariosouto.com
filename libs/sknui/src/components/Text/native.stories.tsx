import { Text } from "./native";

export const Component = Text;

export const controls = {
  props: {
    children: "Sample Text",
    styleSheet: {
      fontSize: 50,
    },
  },
};
