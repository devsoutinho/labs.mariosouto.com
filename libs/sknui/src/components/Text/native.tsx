import React from "react";
import { Text as RNText } from "react-native";
import { StyleSheet } from "../../theme/StyleSheet/native";

interface TextProps {
  styleSheet?: StyleSheet;
  children: React.ReactNode;
}
export function Text({ styleSheet, children }: TextProps) {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  return <RNText style={styleSheet as any}>{children}</RNText>;
}
