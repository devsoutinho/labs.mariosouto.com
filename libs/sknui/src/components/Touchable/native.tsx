import React from "react";
import { TouchableOpacity } from "react-native";
import { StyleSheet } from "../../theme/native";

interface TouchableProps {
  children: React.ReactNode;
  onTap?: () => void;
  disabled?: boolean;
  styleSheet?: StyleSheet;
}
export function Touchable({
  children,
  onTap,
  disabled,
  styleSheet,
}: TouchableProps) {
  return (
    <TouchableOpacity
      onPress={onTap}
      style={
        {
          backgroundColor: disabled ? "red" : "transparent",
          ...styleSheet,
          // eslint-disable-next-line @typescript-eslint/no-explicit-any
        } as any
      }
    >
      {children}
    </TouchableOpacity>
  );
}

Touchable.defaultProps = {};
