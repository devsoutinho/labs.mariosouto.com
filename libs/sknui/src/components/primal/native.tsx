import React from "react";
import { StyleSheet } from "../../theme/StyleSheet/native";

function reduceStyleSheet(styleSheet?: StyleSheet) {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  return Object.entries(styleSheet || {}).reduce<any>((acc, [key, value]) => {
    if (typeof value === "object") {
      return {
        ...acc,
        [key]: value.xs,
      };
    }
    return {
      ...acc,
      [key]: value,
    };
  }, {});
}

interface BoxProps {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  as?: any;
  styleSheet?: StyleSheet;
  children: React.ReactNode;
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  [key: string]: any;
}
export function BoxPrimal({ as, styleSheet, ...props }: BoxProps) {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  const Component = as;
  return <Component style={reduceStyleSheet(styleSheet)} {...props} />;
}
