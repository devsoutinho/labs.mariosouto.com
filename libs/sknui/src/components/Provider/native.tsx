import { ThemeProvider } from "../../theme/components/ThemeProvider";
import { theme } from "../../theme/default/theme";

interface ProviderProps {
  children: React.ReactNode;
}
export function Provider({ children }: ProviderProps) {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
}
