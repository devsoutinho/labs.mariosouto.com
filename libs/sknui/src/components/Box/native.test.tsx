/* eslint-disable @typescript-eslint/no-explicit-any */
import React from "react";
import renderer from "react-test-renderer";

import { Box } from "./native";
import { Text } from "../Text/native";
import { customRender } from "../../../commons/testing-library/native";

// BASE
describe("<Box />", () => {
  it("has 1 child", () => {
    const tree = renderer
      .create(
        <Box>
          <Text>Hi</Text>
        </Box>
      )
      .toJSON();
    const children = (tree as any)?.children;
    expect(children?.length).toBe(1);
  });
});

const render = customRender();

describe("<HomeScreen /> 2", () => {
  it("renders as expected", () => {
    const { toJSON } = render(
      <Box>
        <Text>Hi</Text>
      </Box>
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
