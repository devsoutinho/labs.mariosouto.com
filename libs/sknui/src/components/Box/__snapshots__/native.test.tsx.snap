// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeScreen /> 2 renders as expected 1`] = `
<View
  style={{}}
>
  <Text>
    Hi
  </Text>
</View>
`;
